app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,51 @@
+class Api::V1::ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show update destroy]
+
+  # GET /reservations
+  def index
+    @reservations = current_user.reservations.all
+
+    render json: @reservations
+  end
+
+  # GET /reservations/1
+  def show
+    @reservation = current_user.reservations.find(params[:id])
+
+    render json: @reservation
```

Comment:
As discussed above, you can add the relation with the fitness__activity

i.e `render json: @reservation, status: 200, include: %i[fitness_activity]`